### PR TITLE
read parquet image shard

### PR DESCRIPTION
**Pull request by Le-Zheng**

### CODE REVIEW

**Comment by Le-Zheng** [comment]
jenkins passed: http://10.112.231.51:18888/job/ZOO-PR-Validation/6226/

File: pyzoo/zoo/orca/data/image/parquet_dataset.py [hunk 1]
```diff
@@ -288,20 +288,81 @@ def read_as_tfdataset(path, output_types, output_shapes=None, *args, **kwargs):
...
+        """
+        Creates a `Dataset` that includes only 1/`num_shards` of this dataset.
+        The Dataset will contain all elements of total dataset whose index % num_shards = rank.
+        e.g. the total dataset has 6 chunks, num_shard is 3, the rank 0(worker 0) will contain
+        index 0, 3 of the dataset.
+        """
+
+        def __init__(self, row_group, num_shards=None,
+                     rank=None):
+            super(ParquetDataset).__init__()
```
Comment (yangw1234) [inline pyzoo/zoo/orca/data/image/parquet_dataset.py:309]: Shouldn't it be `super(ParquetIterableDataset)`?

**Comment by yangw1234** [comment]
unit tests may also needed

**Comment by yangw1234** [comment]
@leonardozcm would you mind helping review this pr?

File: pyzoo/zoo/orca/data/image/parquet_dataset.py [hunk 2]
```diff
@@ -138,6 +138,72 @@ def read_as_torch(path):
         raise NotImplementedError()
 
 
+class ParquetIterableDataset(torch.utils.data.IterableDataset):
```
Comment (yangw1234) [inline pyzoo/zoo/orca/data/image/parquet_dataset.py:141]: Is there a way that we can extract these iterator logic into a class not depending on `torch.utils.data.IterableDataset`. 
Becase if we use a pytorch IterableDataset, tensorflow user will also have to install pytorch.
E.g.

```python
class ParquetIterator

    def __init__(self, row_group, shema, num_shards=None, ...):
        pass

    def __next__():

    def __call__():

class ParquetIterableDataset(torch.utils.data.IterableDataset):
       def __init__(...):
             self.iterator = ParquetIterator(...)

       def __next__():
           return self.iterator.__next__()
```

**Comment by leonardozcm** [comment]
You can add torch dataloader ut refer to the old one
https://github.com/leonardozcm/analytics-zoo/blob/fd82deb900bf1096782b4c52b0dd1025a660d0f9/pyzoo/test/zoo/orca/data/test_read_parquet_images.py#L194

File: pyzoo/zoo/orca/data/image/parquet_dataset.py [hunk 3]
```diff
@@ -138,6 +137,70 @@ def read_as_torch(path):
...
 
+class ParquetIterable:
+    """
+    Creates a `Dataset` that includes only 1/`num_shards` of this dataset.
+    The Dataset will contain all elements of total dataset whose index % num_shards = rank.
+    """
+
+    def __init__(self, row_group, schema, num_shards=None,
+                 rank=None, transforms=None):
+        super(ParquetIterable).__init__()
```
Comment (yangw1234) [inline pyzoo/zoo/orca/data/image/parquet_dataset.py:148]: `ParquetIterable` has no ancestor, and this line is not needed, I think.

File: pyzoo/zoo/orca/data/image/parquet_dataset.py [hunk 4]
```diff
@@ -321,61 +385,18 @@ def read_as_dataloader(path, config=None, transforms=None, batch_size=1, *args,
...
-            data_record = []
-            for select_chunk_path in [self.row_group[i] for i in filter_row_group_indexed]:
-                pq_table = pq.read_table(select_chunk_path)
-                df = decode_feature_type_ndarray(pq_table.to_pandas(), schema)
-                data_record.extend(df.to_dict("records"))
-
-            self.datapiece = data_record
-            self.cur = 0
-            self.cur_tail = len(self.datapiece)
+            super(ParquetIterable).__init__()
```
Comment (yangw1234) [inline pyzoo/zoo/orca/data/image/parquet_dataset.py:390]: `super(ParquetIterableDataset).__init__()`?

**Review by yangw1234 — APPROVED** [review]

**Comment by Le-Zheng** [comment]
jenkins passed: http://10.112.231.51:18888/job/ZOO-PR-Validation-Other/6323/

**Pull request merged**